### PR TITLE
fix for bibiserv compilation

### PR DESCRIPTION
**Pull request by pbelmann**
see #6 

@jkrue  can you review this PR?

### CODE REVIEW

File: bibimainapp/pom.xml [hunk 1]
```diff
@@ -20,48 +21,59 @@
...
-        
+
         <dependency>
             <groupId>${project.groupId}</groupId>
             <artifactId>bibiservsearch</artifactId>
-            <version>1.1</version>        
+            <version>1.1</version>
+        </dependency>
+
+        <dependency>
```
Comment (jkrue) [inline bibimainapp/pom.xml:41]: Why we need an extra dependency to an older Spring Security version (3.0.1 instead if 3.0.8) ?

File: bibimainapp/pom.xml [hunk 2]
```diff
@@ -20,48 +21,59 @@
...
         </dependency>
-        
-        
+
         <!-- https://mvnrepository.com/artifact/com.sun.jersey/jersey-server -->
         <dependency>
-            <groupId>com.sun.jersey</groupId>
-            <artifactId>jersey-server</artifactId>
-            <version>1.2</version>
+            <groupId>org.glassfish.jersey.containers</groupId>
```
Comment (jkrue) [inline bibimainapp/pom.xml:54]: Updating Jersey from 1.2 to 2.1 leads to a not working REST API for deployed application during Runtime. However the build in REST API (/manager/**) seem to be working fine with the new release.

File: bibimainapp/src/main/java/de/unibi/cebitec/bibiserv/web/rest/BiBiServRESTServlet.java [hunk 3]
```diff
@@ -4,15 +4,16 @@
  */
 package de.unibi.cebitec.bibiserv.web.rest;
 
-import com.sun.jersey.spi.container.servlet.ServletContainer;
 import de.unibi.cebitec.bibiserv.server.manager.Core;
+import org.glassfish.jersey.servlet.ServletContainer;
```
Comment (jkrue) [inline bibimainapp/src/main/java/de/unibi/cebitec/bibiserv/web/rest/BiBiServRESTServlet.java:8]: The BiBiServRestServelet is used by BiBiServ tools which are not part of the bibimainapp. Updating Jersey to version 2.1 leads to a not working REST API for attached/deployed tools.

File: schemas/alignmentml/pom.xml [hunk 4]
```diff
@@ -49,6 +49,91 @@
...
+                                <available property="isFile" type="file" file="${de.unibi.cebitec.bibiserv:biotypes:jar}"/>
+                            </target>
+                            <exportAntProperties>true</exportAntProperties>
+                        </configuration>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>1.1</id>
```
Comment (jkrue) [inline schemas/alignmentml/pom.xml:66]: Ok!

File: schemas/bibiservabstraction/pom.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>       
-        <xjc.arguments>${de.unibi.cebitec.bibiserv:jnlp:jar} ${de.unibi.cebitec.bibiserv:unserhtml:jar}</xjc.arguments>
+        <xjc.arguments>${de.unibi.cebitec.bibiserv:jnlp:jar},${de.unibi.cebitec.bibiserv:unserhtml:jar}</xjc.arguments>
```
Comment (jkrue) [inline schemas/bibiservabstraction/pom.xml:12]: Ok!

File: schemas/rnastructalignmentml/pom.xml [hunk 6]
```diff
@@ -49,6 +49,91 @@
...
+                            </target>
+                            <exportAntProperties>true</exportAntProperties>
+                        </configuration>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>1.1</id>
+                        <phase>generate-sources</phase>
```
Comment (jkrue) [inline schemas/rnastructalignmentml/pom.xml:67]: OK!

File: schemas/rnastructml/pom.xml [hunk 7]
```diff
@@ -49,6 +49,91 @@
                 <version>1.7</version>
                 <executions>
                     <execution>
+                        <id>1</id>
```
Comment (jkrue) [inline schemas/rnastructml/pom.xml:52]: Ok!

**Review by jkrue — CHANGES_REQUESTED** [review]
This Pull request deals with two different targets:
1.  Building the whole project using the parent pom.
   The current release breaks when compile some of schemas/*\* projects (XJC dependency), but only when calling the parent pom. Everything is fine when building them standalone.
2. Update Jersey from 1.2 to 2.1
   The BiBiServRestServlet (which is mainly affected by this update) is used by BiBiServ tools which are not part of the bibimainapp project. Updating Jersey to version 2.1 leads to a non working REST API for attached/deployed tools. Is there a special resaon why upgrading Jersey ? The 'before-moving-to-mvn' version works fine with the older version running on GF 3.x and 4.x (which are the preferred Application server for the bibiserv)

However I suggest to merge all parts belonging to 1) asap and  do some more test  before merge 2)

**Comment by pbelmann** [comment]
@jkrue  thanks for the review. I agree that we should first merge the fix  for the schema projects and create a separate issue for upgrading jersey which seems to be more complicated. I will close this PR and create a new one.

**Pull request closed without merging**